src/main/java/zd/zero/waifu/motivator/plugin/assets/RemoteAssetManager.kt

```diff
@@ -68,8 +68,8 @@ abstract class RemoteAssetManager<T : AssetDefinition, U : Asset>(
         localAssets
 
     fun supplyRemoteAssetDefinitions(): List<T> =
-        remoteAndLocalAssets.filterNot { remoteAndLocal ->
-            localAssets.toList().any { it.path == remoteAndLocal.path }
+        remoteAndLocalAssets.filterNot { remoteOrLocalAsset ->
+            localAssets.contains(remoteOrLocalAsset)
```

Comment:
Kotlin [data classes](https://kotlinlang.org/docs/reference/data-classes.html#data-classes) have built in hash-code/equals. So we can just use the `Set` API.

Reply:
Right! I just missed streams 😂